codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/ShapeValueGenerator.java

```diff
@@ -0,0 +1,358 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ *
+ */
+
+package software.amazon.smithy.go.codegen;
+
+import java.util.logging.Logger;
+import software.amazon.smithy.codegen.core.CodegenException;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.node.ArrayNode;
+import software.amazon.smithy.model.node.BooleanNode;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.NodeVisitor;
+import software.amazon.smithy.model.node.NullNode;
+import software.amazon.smithy.model.node.NumberNode;
+import software.amazon.smithy.model.node.ObjectNode;
+import software.amazon.smithy.model.node.StringNode;
+import software.amazon.smithy.model.shapes.CollectionShape;
+import software.amazon.smithy.model.shapes.MapShape;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeType;
+import software.amazon.smithy.model.shapes.StructureShape;
+import software.amazon.smithy.model.traits.EnumTrait;
+
+public class ShapeValueGenerator {
+    private static final Logger LOGGER = Logger.getLogger(ShapeValueGenerator.class.getName());
+
+    protected final Model model;
+    protected final SymbolProvider symbolProvider;
+
+    public ShapeValueGenerator(Model model, SymbolProvider symbolProvider) {
+        this.model = model;
+        this.symbolProvider = symbolProvider;
+    }
+
+    public void writeShapeValueInline(GoWriter writer, Shape shape, Node params) {
+        switch (shape.getType()) {
+            case STRUCTURE:
+                structDeclShapeValue(writer, shape.asStructureShape().get(), () -> {
+                    params.accept(new ShapeValueNodeVisitor(writer, this, shape));
+                });
+                break;
+
+            case SET:
+            case LIST:
+                listDeclShapeValue(writer, (CollectionShape) shape, () -> {
+                    params.accept(new ShapeValueNodeVisitor(writer, this, shape));
+                });
+                break;
+
+            case MAP:
+                mapDeclShapeValue(writer, shape.asMapShape().get(), () -> {
+                    params.accept(new ShapeValueNodeVisitor(writer, this, shape));
+                });
+                break;
+
+            case UNION:
+            case DOCUMENT:
+                writer.writeInline("/* union and document not supported */");
+                break;
+
+            default:
+                scalarWrapShapeValue(writer, shape, () -> {
+                    params.accept(new ShapeValueNodeVisitor(writer, this, shape));
+                });
+        }
+    }
+
+    public void structDeclShapeValue(GoWriter writer, StructureShape shape, Runnable inner) {
```

Comment:
Do these need to be public?

Reply:
No, can move these to default scope, or protected.